torchbenchmark/util/model.py

```diff
@@ -483,7 +483,7 @@ def enable_amp(self):
             self.amp_context = lambda: torch.cpu.amp.autocast()
         elif self.device == "cuda":
             self.amp_context = lambda: torch.cuda.amp.autocast()
-        if is_staged_train_test(self):
+        if self.test == "train" and is_staged_train_test(self):
             self.forward_contexts.append(self.amp_context)
```

Comment:
I think we should handle `self.test == "eval"` here as well?
For "eval" test, we should just add it to `self.run_contexts`.

Reply:
yes, you are right. 
while i modify it, i have a question, do we need to assert error if self.test is train and is_staged_train_test is false, i see that in the amp_fp16 path, it asserts error. please refer to below code.
https://github.com/pytorch/benchmark/blob/main/torchbenchmark/util/extra_args.py#L189C9-L198

Reply:
Either assert error or print a warning in this case.
The reason is because usually models do not want to enable amp in the backward pass, only forward pass.

I am leaning towards print a warning message in both `--amp` and `--amp_fp16`, what do you think?

Reply:
you are right, I added the warning, please take a look.